generate/canoto.go

```diff
@@ -931,56 +922,58 @@ func makeValid(m message) string {
 func makeZeroOneOfCache(m message) string {
 	var s strings.Builder
 	for _, oneOf := range m.OneOfs() {
-		_, _ = fmt.Fprintf(&s, "\tc.canotoData.%sOneOf = 0\n", oneOf)
+		_, _ = fmt.Fprintf(&s, "\tc.canotoData.%sOneOf.Store(0)\n", oneOf)
```

Comment:
We shouldn't zero out these values here. we should calculate their correct values (which may be 0) and then set them at the end of the function.